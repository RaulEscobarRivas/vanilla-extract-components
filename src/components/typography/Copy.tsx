import React from "react";
import { useAsProp } from "../../hooks/useAsProp";
import { useCopyStyles } from "./useCopyStyles";
import { CopyProps } from "./types";

export function Copy({
  bold,
  color = "gray.800",
  ellipsis,
  extraLight,
  shadow,
  size = "md",
  textAlign,
  ...rest
}: CopyProps) {
  const [Tag, props] = useAsProp(
    { as: "p", ...rest },
    {
      className: useCopyStyles({
        bold,
        color,
        ellipsis,
        extraLight,
        shadow,
        size,
        textAlign,
      }),
    }
  );

  return <Tag {...props} />;
}
