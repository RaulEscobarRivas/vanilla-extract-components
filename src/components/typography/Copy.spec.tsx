import React from "react";
import { render } from "@testing-library/react";
import { Copy } from "./Copy";

describe("Copy", () => {
  it("should render default style", () => {
    const { container } = render(<Copy />);

    expect(container).toMatchSnapshot();
  });

  it("should render custom span tag", () => {
    const { container } = render(<Copy as="span" />);

    expect(container).toMatchSnapshot();
  });

  it("should render size style", () => {
    const { container } = render(<Copy size="xs" />);

    expect(container).toMatchSnapshot();
  });

  it("should render shadow style", () => {
    const { container } = render(<Copy shadow />);

    expect(container).toMatchSnapshot();
  });

  it("should render bold style", () => {
    const { container } = render(<Copy bold />);

    expect(container).toMatchSnapshot();
  });

  it("should render ellipsis style", () => {
    const { container } = render(<Copy ellipsis />);

    expect(container).toMatchSnapshot();
  });

  it("should render extraLight style", () => {
    const { container } = render(<Copy extraLight />);

    expect(container).toMatchSnapshot();
  });

  it("should render textAlign=left style", () => {
    const { container } = render(<Copy textAlign="left" />);

    expect(container).toMatchSnapshot();
  });

  it("should render textAlign=right style", () => {
    const { container } = render(<Copy textAlign="right" />);

    expect(container).toMatchSnapshot();
  });

  it("should render textAlign=center style", () => {
    const { container } = render(<Copy textAlign="center" />);

    expect(container).toMatchSnapshot();
  });

  it("should render color style", () => {
    const { container } = render(<Copy color="green.300" />);

    expect(container).toMatchSnapshot();
  });

  it("should render variant styles", () => {
    const { container } = render(
      <Copy
        as="p"
        color="white"
        bold
        shadow
        ellipsis
        textAlign="right"
        size="xl"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
