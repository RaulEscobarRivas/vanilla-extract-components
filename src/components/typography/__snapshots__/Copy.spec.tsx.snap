// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy should render bold style 1`] = `
<div>
  <p
    class="Copy_copyBase__o5k5655 variants_colorStyle_gray.800__nc0lsl4 Copy_sizeStyle_md__o5k5652 variants_boldStyle__nc0lslg"
  />
</div>
`;

exports[`Copy should render color style 1`] = `
<div>
  <p
    class="Copy_copyBase__o5k5655 variants_colorStyle_green.300__nc0lsl7 Copy_sizeStyle_md__o5k5652"
  />
</div>
`;

exports[`Copy should render custom span tag 1`] = `
<div>
  <span
    class="Copy_copyBase__o5k5655 variants_colorStyle_gray.800__nc0lsl4 Copy_sizeStyle_md__o5k5652"
  />
</div>
`;

exports[`Copy should render default style 1`] = `
<div>
  <p
    class="Copy_copyBase__o5k5655 variants_colorStyle_gray.800__nc0lsl4 Copy_sizeStyle_md__o5k5652"
  />
</div>
`;

exports[`Copy should render ellipsis style 1`] = `
<div>
  <p
    class="Copy_copyBase__o5k5655 variants_colorStyle_gray.800__nc0lsl4 Copy_sizeStyle_md__o5k5652 variants_ellipsisStyle__nc0lsle"
  />
</div>
`;

exports[`Copy should render extraLight style 1`] = `
<div>
  <p
    class="Copy_copyBase__o5k5655 variants_colorStyle_gray.800__nc0lsl4 Copy_sizeStyle_md__o5k5652 variants_extraLightStyle__nc0lslf"
  />
</div>
`;

exports[`Copy should render shadow style 1`] = `
<div>
  <p
    class="Copy_copyBase__o5k5655 variants_colorStyle_gray.800__nc0lsl4 Copy_sizeStyle_md__o5k5652 variants_shadowStyle__nc0lsld"
  />
</div>
`;

exports[`Copy should render size style 1`] = `
<div>
  <p
    class="Copy_copyBase__o5k5655 variants_colorStyle_gray.800__nc0lsl4 Copy_sizeStyle_xs__o5k5650"
  />
</div>
`;

exports[`Copy should render textAlign=center style 1`] = `
<div>
  <p
    class="Copy_copyBase__o5k5655 variants_colorStyle_gray.800__nc0lsl4 variants_textAlignStyle_center__nc0lslj Copy_sizeStyle_md__o5k5652"
  />
</div>
`;

exports[`Copy should render textAlign=left style 1`] = `
<div>
  <p
    class="Copy_copyBase__o5k5655 variants_colorStyle_gray.800__nc0lsl4 variants_textAlignStyle_left__nc0lslh Copy_sizeStyle_md__o5k5652"
  />
</div>
`;

exports[`Copy should render textAlign=right style 1`] = `
<div>
  <p
    class="Copy_copyBase__o5k5655 variants_colorStyle_gray.800__nc0lsl4 variants_textAlignStyle_right__nc0lsli Copy_sizeStyle_md__o5k5652"
  />
</div>
`;

exports[`Copy should render variant styles 1`] = `
<div>
  <p
    class="Copy_copyBase__o5k5655 variants_colorStyle_white__nc0lsla variants_textAlignStyle_right__nc0lsli Copy_sizeStyle_xl__o5k5654 variants_shadowStyle__nc0lsld variants_ellipsisStyle__nc0lsle variants_boldStyle__nc0lslg"
  />
</div>
`;
